test/e2e/kube-ovn/ipam/ipam.go

```diff
@@ -509,4 +509,56 @@ var _ = framework.Describe("[group:ipam]", func() {
 		deploy = deployClient.PatchSync(deploy, patchedDeploy)
 		checkFn()
 	})
+
+	framework.ConformanceIt("should allocate right IPs for the statefulset when there are multiple IP Pools added to its namespace", func() {
+		replicas := 1
+		ipsCount := 12
+		ipsRange1 := framework.RandomIPPool(cidr, ipsCount)
+		ipsRange2 := framework.RandomIPPool(cidr, ipsCount)
+		testStsName := "test-statefulset"
+		testSubnetName := "ip-pool-subnet2"
+		testIPPool1Name := "ip-pool1"
+		testIPPool2Name := "ip-pool2"
+
+		ginkgo.By("Creating a new subnet " + testSubnetName)
+		testCidr := framework.RandomCIDR(f.ClusterIPFamily)
+		testSubnet := framework.MakeSubnet(testSubnetName, "", testCidr, "", "", "", nil, nil, []string{namespaceName})
+		subnetClient.CreateSync(testSubnet)
```

Comment:
```suggestion
		testSubnet := framework.MakeSubnet(testSubnetName, "", testCidr, "", "", "", nil, nil, []string{namespaceName})
		ginkgo.DeferCleanup(func() {
			ginkgo.By("Deleting subnet " + testSubnetName)
			subnetClient.DeleteSync(testSubnetName)
		})
		subnetClient.CreateSync(testSubnet)
```

Let's use `ginkgo.DeferCleanup()` to ensure test resources are cleaned up.